src/private_broadcast.cpp

```diff
@@ -14,10 +15,79 @@ void PrivateBroadcast::Add(const CTransactionRef& tx) EXCLUSIVE_LOCKS_REQUIRED(!
     }
 }
 
+std::optional<CTransactionRef> PrivateBroadcast::GetTxForBroadcast() EXCLUSIVE_LOCKS_REQUIRED(!m_mutex)
+{
+    LOCK(m_mutex);
+    if (m_by_priority.empty()) {
+        return std::nullopt;
+    }
+    const Txid& txid = m_by_priority.begin()->second;
+    auto it = m_by_txid.find(txid);
+    if (Assume(it != m_by_txid.end())) {
+        return it->second.tx;
+    }
+    m_by_priority.erase(m_by_priority.begin());
+    return std::nullopt;
+}
+
+void PrivateBroadcast::PushedToNode(const NodeId& nodeid, const Txid& txid) EXCLUSIVE_LOCKS_REQUIRED(!m_mutex)
+{
+    LOCK(m_mutex);
+    m_by_nodeid.emplace(nodeid, txid);
+}
+
+bool PrivateBroadcast::FinishBroadcast(const NodeId& nodeid, bool confirmed_by_node) EXCLUSIVE_LOCKS_REQUIRED(!m_mutex)
+{
+    LOCK(m_mutex);
+    auto it = m_by_nodeid.find(nodeid);
+    if (it == m_by_nodeid.end()) {
+        return false;
+    }
+    const Txid txid{it->second};
+    m_by_nodeid.erase(it);
+
+    if (!confirmed_by_node) {
+        return true;
+    }
+
+    // Update broadcast stats.
+
+    auto iters = Find(txid);
+    if (!iters) {
+        return true;
+    }
+    Priority& priority = iters->by_txid->second.priority;
+
+    ++priority.num_broadcasted;
+    priority.last_broadcasted = GetTime<std::chrono::microseconds>();
```

Comment:
b81c3cc19ec8055d04b38a72fdfabbfcb2ed2358

Isn't this `GetTime()` deprecated?

https://github.com/bitcoin/bitcoin/blob/7a7e7d189dfe5efa6ab3c644f55f410412163503/src/util/time.h#L95-L100

Reply:
Replaced with `NodeClock::now()`.